java/client/src/org/openqa/selenium/support/ui/ExpectedConditions.java

```diff
@@ -348,10 +348,12 @@ public String toString() {
                                                                     final String text) {
 
     return new ExpectedCondition<Boolean>() {
+      private String elementText = null;
```

Comment:
Each `ExpectedCondition` implements `java.util.Function` These are expected to be stateless. This condition will leak previous `elementText` on the second usage, which doesn't seem ideal.